zerver/tests/test_message_fetch.py

```diff
@@ -245,12 +245,16 @@ def test_add_term_using_is_operator_non_dm_operand_and_negated(self) -> None:  #
         self._do_add_term_test(term, where_clause, params)
 
         term = dict(operator="is", operand="mentioned", negated=True)
-        where_clause = "WHERE NOT ((flags & %(flags_1)s) != %(param_1)s OR (flags & %(flags_2)s) != %(param_2)s)"
+        where_clause = "WHERE NOT ((flags & %(flags_1)s) != %(param_1)s OR (flags & %(flags_2)s) != %(param_2)s OR (flags & %(flags_3)s) != %(param_3)s OR (flags & %(flags_4)s) != %(param_4)s)"
```

Comment:
We should mainly discuss this in the `models.py` code, so feel free to mark this comment as resolved, but seeing this has been considering that we might want to write it as a single `((flags & {mention_flags_mask}))` where the mask is a bitword with the 4 relevant flags 1 and the rest 0s.